connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -93,6 +96,19 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
       transformRelation(rel.getInput))
   }
 
+  private def transformDataFrameFunction(rel: proto.DataFrameFunction): LogicalPlan = {
+    val child = transformRelation(rel.getInput)
+
+    rel.getFunctionCase match {
+      case proto.DataFrameFunction.FunctionCase.SUMMARY =>
+        StatFunctions
+          .summary(Dataset.ofRows(session, child), rel.getSummary.getStatisticsList.asScala.toSeq)
```

Comment:
This is fine for now but it's going to truncate the SQL plans that disable further optimization. We should probably add dedicated plans for `def summary` in `Dataset` itself.

For now, LGTM

Reply:
yes, then it will have more optimization space. let us add new plan for it. thanks

Reply:
+1! 

I don't know how to add a new plan. It would be very useful to have a PR as an example.

Reply:
cc @cloud-fan @HyukjinKwon 
since we had reimplemented the `df.summary` https://github.com/apache/spark/commit/6a0713a141fa98d83029d8388508cbbc40fd554e, are there some differences in sql optimization between this method (directly invoke `df.summary`) and adding a dedicated plan?

Reply:
Some rules may not work as they don't recognize the new plan.

Reply:
what do you mean by `truncate the SQL plans`? DataFrame transformations just accumulate the logical plan.

Reply:
@cloud-fan the old `df.summary` eagerly compute the statistics and always return a `LocalRelation`

Reply:
Oh that's an issue. Can it be solved by updating `df.summary` implementation?

Reply:
yes, it has been resolved

Reply:
In the new impl